collect_app/src/main/java/org/odk/collect/android/utilities/TextUtils.java

```diff
@@ -69,20 +69,31 @@ private TextUtils() {
 
     }
 
-    protected static String markdownToHtml(String text) {
+    public static String markdownToHtml(String text) {
 
         text = text.replaceAll("<([^a-zA-Z/])", "&lt;$1");
         // https://github.com/enketo/enketo-transformer/blob/master/src/markdown.js
 
         // span - replaced &lt; and &gt; with <>
         text = ReplaceCallback.replace("(?s)<\\s?span([^\\/\n]*)>((?:(?!<\\/).)+)<\\/\\s?span\\s?>",
                 text, createSpan);
+
+        //intermediary replacements keys for special characters, N/B: These symbols are not meant to be interpreted as markdown
+        text = text.replaceAll("(?s)\\\\#", "==");
```

Comment:
Now this looks like the right place for those replacements! You should be able to change them directly to their HTML escape sequences (https://www.w3.org/MarkUp/html-spec/html-spec_13.html). Using these intermediary replacement keys works but it means that those character sequences now can't be used in a form. Going straight to HTML sequences means no new characters need special meaning.

Reply:
Okay.
On it